org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/InitHandler.java

```diff
@@ -93,6 +93,15 @@ public InitHandler(ProjectsManager manager, PreferenceManager preferenceManager,
 
 	@Override
 	public Map<?, ?> handleInitializationOptions(InitializeParams param) {
+		// https://github.com/redhat-developer/vscode-java/issues/3184
+		// start the m2e and buildship plugin before calling JavaCore.setOptions
```

Comment:
Which `JavaCore.setOptions(..)` are you refering to ? There is call to that as soon as plugin activation happens for jdt.ls.core. See https://github.com/eclipse/eclipse.jdt.ls/blob/a1e091430705860a95456a7332a223d594025716/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/PreferenceManager.java#L160 . This change doesn't happen before that..

Reply:
See https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/PreferenceManager.java#L222

It is called from 
InitHandler.handleInitializationOptions(InitializeParams) > BaseInitHandler.handleInitializationOptions(InitializeParams) > PreferenceManager.update(Preferences)

Reply:
> There is call to that as soon as plugin activation happens for jdt.ls.core.

This is not an issue because no project is imported at that moment.